src/views/document/utils/boxes/RecentOutingsBox.vue

```diff
@@ -3,23 +3,25 @@
     <div class="title is-2">
       <span v-if="documentType === 'image'" v-translate>Associated outings</span>
       <span v-else v-translate>Last outings</span>
-      <router-link
-        :to="{ name: 'outings', query: query }"
-        class="is-size-5"
-        v-if="outings.length !== 0 && !hideSeeAllResultsButton"
-        v-translate
-      >
-        show all
-      </router-link>
     </div>
 
     <div v-for="(outing, i) of outings" :key="i">
       <pretty-outing-link :outing="outing" />
     </div>
 
-    <div v-if="documentType == 'route'" class="has-text-centered add-section">
-      <add-link document-type="outing" :query="query" class="button is-primary">
-        <span v-if="outings.length === 0" v-translate> Add the first outing </span>
+    <div
+      v-if="(!hideSeeAllResultsButton && outings.length) || documentType == 'route'"
+      class="has-text-centered add-section"
+    >
+      <router-link
+        :to="{ name: 'outings', query: query }"
+        class="button is-primary"
+        v-if="outings.length && !hideSeeAllResultsButton"
```

Comment:
should be "!hideSeeAllResultsButton && outings.length" for performance & consistency

Reply:
I don't think retrieving an array's length has a bigger cost than checking a boolean? It's a simple address lookup. OK for consistency of reading though.